### ISSUE TITLE
"View In Object Browser" does nothing for packages

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-project-system#1256]
1. Create a Console App (.NET Core)
2. Expand Dependencies -> NuGet
3. Right-click on Microsoft.NETCore.App and choose View In Object Browser

-- Nothing --

I suspect we want to disable this, unless we support opening all of it's assemblies in the object browser.

**Comment by abpiskunov** [comment]
dupe/related of this one https://github.com/dotnet/roslyn-project-system/issues/403

**Comment by davkean** [comment]
I don't think this is a dupe, at least in the assembly/project version we should just use the existing infrastructure to open them. In this case, what would we even do? Loop over every reference and view them?

**Comment by abpiskunov** [comment]
sure. For package i was planning to hide it as part of #403. Let's keep this issue to split the work and simplify reviews :)

**Comment by abpiskunov** [comment]
Note: after cross targeting dependencies change , View in object browser needs to be handled on roslyn side completely for any kind of dependencies: assemblies and project we support it, for packages don't show context menu at all.   CPS implementation of this command really hardcoded to IProjectItemTree , but we have IProjectItemTree items only for active target , not for others.  

We need 
- handle command ourselves to control visibility http://index/?rightProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&file=Package%5cComponents%5cDefaultCommandSet2KHandler.cs&line=133
- handle actual command execution to show assembly or project in Object browser  http://index/?rightProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&file=Package%5cComponents%5cDefaultVsUIHierarchyWindowCmdsHandler.cs&line=175

**Comment by BenjaminHolland** [comment]
@abpiskunov The links provided are broken.

**Comment by abpiskunov** [comment]
Find files and lines from links in actual CPS code instead of index please.

Thanks,
Anton

On Jun 18, 2017, at 12:59 PM, Ben Holland <notifications@github.com<mailto:notifications@github.com>> wrote:


@abpiskunov<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fabpiskunov&data=02%7C01%7Canton.piskunov%40microsoft.com%7C6d89ff6b8ed74a77aedc08d4b684837e%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636334127640906397&sdata=T9SlAkBqFQq%2FbS3pfvWrO7cF2AR4JE7im6OUnn7HaP8%3D&reserved=0> The links provided are broken.

**Comment by davkean** [comment]
@BenjaminHolland These are internal links to non-OSS.

**Comment by reyou** [comment]
This is still and issue for aspdotnetcore projects, and it just does not navigate.

It works for first 5-10, navigation, then after adding new classes etc, it stops working.

any improvement or workaround for this?

**Comment by fmicle** [comment]
This is an issue for me in Visual Studio 2019 Preview. I can browse the NuGet dependency fine in VS 2017, but nothing happens in 2019.

**Comment by fmicle** [comment]
[Screenshot](https://imgur.com/a/R8XDbjP)

**Comment by capinredbeard** [comment]
I have this same issue in VS 2019.

**Comment by v-kydela** [comment]
2.5 years and many Visual Studio versions later and this serious bug still has not been addressed

**Comment by 18549** [comment]
How are we supposed to determine the exports of a third party assembly, ildasm?

**Comment by drewnoakes** [comment]
Legacy does not include the _View in Object Browser_ context menu item for package references.

When using the old-style packages.config, direct assembly references were added to the project as top-level items, and these would support the object browser.

Unlike legacy, the new project system allows expanding package references to see their inner assemblies. It may be possible to navigate to the object browser from a child assembly directly.

**Comment by JohannSig** [comment]
I'm trying to view the contents of a referenced DLL in an ASP.NET Core 3.1 project in Visual Studio 2019. This used to be done with Object Browser, but that gives me no information at all (not even about my own project). Has this feature been dropped in Core?

**Comment by drewnoakes** [comment]
@JohannSig the lack of this feature is what this issue is tracking.

In 16.7 we will show considerably more info in the tree for all dependencies, including packages. Each assembly in the package will be visible. We want to add this feature to those nodes.

**Comment by ShiningMassXAcc** [comment]
I have "Version 16.7.0 Preview 3.1" of VS.  I cannot expand a package reference added nuget package, and get the following error (even after building / install).

![image](https://user-images.githubusercontent.com/43765765/87100720-018d9300-c202-11ea-8a92-6aa3d6b42db3.png)

![image](https://user-images.githubusercontent.com/43765765/87100795-2a158d00-c202-11ea-81be-2ae31be3012f.png)

**Comment by drewnoakes** [comment]
@ShiningMassXAcc from your screenshot I see that you're using the 'legacy' project system (it has a "References" node, not a "Dependencies" node).

The improved behaviour is in the .NET Project System (which lives in this repo).

Expandable packages are provided by this project system only. To use it, you would update your projects to be SDK-style.

Note that "View in Object Browser" for those assemblies is still not implemented in this project system.
